packages/SwingSet/src/kernel/kernel.js

```diff
@@ -503,23 +510,93 @@ export default function buildKernel(kernelEndowments) {
     });
 
     // the vatManager invokes setup() to build the userspace image
-    const manager = makeVatManager(
+    return makeVatManager(
       vatID,
       syscallManager,
       setup,
       helpers,
       kernelKeeper,
-      vatKeeper,
-    );
-    ephemeral.vats.set(
-      vatID,
-      harden({
-        manager,
-        enablePipelining: Boolean(options.enablePipelining),
-      }),
+      kernelKeeper.provideVatKeeper(vatID),
     );
   }
 
+  // This kernel (and the genesis vats) will have been evaluated in a context in
+  // which these canonical symbols were available (which is why the require
+  // statements here work). That was achieved by running makeRequire() in
+  // controller.js. We want to provide the same context when we evaluate dynamic
+  // vats.
+  function kernelRequire(nameArg) {
+    const name = `${nameArg}`;
+    const HARDEN = '@agoric/harden';
+    const NAT = '@agoric/nat';
+    const EVALUATE = '@agoric/evaluate';
+    switch (name) {
+      case HARDEN:
+        // eslint-disable-next-line global-require,import/no-dynamic-require
+        return require(HARDEN);
```

Comment:
I'm not sure we should do this now, but in a future cleanup I'm wondering if we should add a `import evaluate from '@agoric/evaluate'; import Nat from '@agoric/nat';` at the top of this file, and remove the `require()` calls from this function (so return `harden` and `Nat` and `evaluate`). Then in `createVatDynamically` we could just call `evaluate.evaluateProgram(buildFnSrc, req)` instead of using the generated `require()` function to extract the evaluator.

Reply:
I guess the other issue is that it'd be nice for all swingset code to be ES6 modules, with `import`, rather than a sort of hybrid combination of `import` and `require`. The way we currently use `rollup` to convert the kernel files into a single string means that we can tolerate the presence of `require`, but I expect that to change in the new SES API (and which is already different to run under XS), and `require` will be forbidden.

Let's land this without that change, but let's make a note (#444) to fix it up in a subsequent patch.